### DISCUSSION TITLE
First time forking a repo, am I doing it right? (Regarding ettiquettes and conventions)

### DISCUSSION THREAD

**Discussion by lumeriith** [community/community#22148] [New to GitHub]
<p>Hello, I’ve been developing some games with Unity. In several projects, I’ve been using a modified version of an open source project “QuickOutline” on GitHub, making it run on a separate thread to not block the main thread.</p>
<p>I thought my modifications would be helpful to some people, so I’m thinking I’ll fork the repo and change the name to “ThreadedQuickOutline” adding features and also fixing some bugs too.</p>
<p>Am I doing this right? Is it okay (or ethical?) to change the name?<br>
Also should I move original Readme.md somewhere else and put mine explaining why my repo exists and what it does? If not, what should I do to state such information?</p>
<p>The original project is licensed under MIT License and it hasn’t been maintained for a couple of years.</p>
<p>Thanks in advance for helping out a complete newb, &lt;3</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Lumeriith" data-post="1" data-topic="119545">
<div class="title">
<div class="quote-controls"></div>
 Lumeriith:</div>
<blockquote>
<p>Am I doing this right? Is it okay (or ethical?) to change the name?</p>
</blockquote>
</aside>
<p>Changing the name is a good idea, because it avoids people confusing your fork with the older project. There are quite a few open source projects around with name suffixes like “-ng” or “2” because they were forked from inactive projects and the developers wanted to show that they are forks. Being more creative about the name certainly won’t hurt. :wink:</p>
<aside class="quote no-group" data-username="Lumeriith" data-post="1" data-topic="119545">
<div class="title">
<div class="quote-controls"></div>
 Lumeriith:</div>
<blockquote>
<p>Also should I move original Readme.md somewhere else and put mine explaining why my repo exists and what it does? If not, what should I do to state such information?</p>
</blockquote>
</aside>
<p>Keeping a note about what you forked from in the Readme is definitely a good idea. If the old code is still around a link is good, too. Whether you do that by editing the old Readme or creating your own mostly depends on whether the old content still makes sense for your changed version (e.g. if it has old usage information that doesn’t apply to your fork any more, there’s little point in keeping it).</p>
<p>It is important that you keep information on the origin of code and its licensing as described in the license itself. Shouldn’t be difficult, but I am not a lawyer.</p>
